identity/flask.py

```diff
@@ -18,19 +18,56 @@
 
 class Auth(WebFrameworkAuth):
     """A long-live identity auth helper for a Flask web project."""
+    _app_initialized = False
 
     def __init__(self, app: Flask, *args, **kwargs):
         """Create an identity helper for a web application.
 
         :param Flask app:
-            A Flask app instance. It will be used to register the routes.
+            It can be a Flask app instance, or None.
+
+            1. If your app object is globally available, you may pass it in here.
+               Usage::
+
+                # In your app.py
+                app = Flask(__name__)
+                auth = Auth(app, authority=..., client_id=..., ...)
+
+            2. But if you are using `Application Factory pattern
+            <https://flask.palletsprojects.com/en/latest/patterns/appfactories/>`_,
+            your app is not available globally, so you need to pass ``None`` here,
+            and call :func:`Auth.init_app()` later,
+            inside or after your app factory function. Usage::
+
+                # In your auth.py
+                auth = Auth(None, authority=..., client_id=..., ...)
```

Comment:
```suggestion
                auth = Auth(app=None, authority=..., client_id=..., ...)
```

I think that looks nicer when passing in None